lib/model/folder.go

```diff
@@ -141,19 +141,21 @@ func (f *folder) deleteDir(path string, file protocol.FileInfo, matcher *ignore.
 	return osutil.InWritableDir(os.Remove, realName)
 }
 
-// deletefile attempts to delete the given file
+// deleteFile attempts to delete the given file
 // Takes sync conflict and versioning settings into account
-func (f *folder) deleteFile(path string, file protocol.FileInfo, ver versioner.Versioner, maxConflicts int) error {
+func (f *folder) deleteFile(folderPath string, file protocol.FileInfo, ver versioner.Versioner, maxConflicts int) error {
 	var err error
 
-	realName, err := rootedJoinedPath(path, file.Name)
+	realName, err := rootedJoinedPath(folderPath, file.Name)
 	if err != nil {
 		return err
 	}
 
-	if maxConflicts > 0 {
+	cur, ok := f.model.CurrentFolderFile(f.folderID, file.Name)
+	if ok && f.inConflict(cur.Version, file.Version) && maxConflicts > 0 {
 		// There is a conflict here. Move the file to a conflict copy instead
 		// of deleting.
+		file.Version = file.Version.Merge(cur.Version)
```

Comment:
This unfortunately doesn't accomplish what it needs to, given that this `file` doesn't get passed into the database in the end. That'll leave other devices without the conflict resolved, causing another conflict when they get our change.